packages/experimental/Menu/SPEC.md

```diff
@@ -0,0 +1,345 @@
+
+## Sample code
+
+The below samples do not represent the definitive props of the final implemented component, but represent the ideal final implementations. Can be subject to change during the implementation phase.
+
+### Basic Menu
+
+```tsx
+const menu = (
+  <Menu>
+    <MenuTrigger><Button>Open menu</Button></MenuTrigger>
+    <MenuPopover>
+      <MenuList>
+        <MenuItem>Option 1</MenuItem>
+        <MenuItem>Option 2</MenuItem>
+        <MenuItem>Option 3</MenuItem>
+      </MenuList>
+    </MenuPopover>
+  <Menu>
+)
+```
+
+### Submenus
+
+```tsx
+const menu = (
+  <Menu>
+    <MenuTrigger><Button>Open menu</Button></MenuTrigger>
+    <MenuPopover>
+      <MenuList>
+        <MenuItem>Option 1</MenuItem>
+        <Menu>
+          <MenuTrigger>
+            <MenuItem>Open submenu</MenuItem>
+          </MenuTrigger>
+          <MenuPopover>
+            <MenuList>
+              <MenuItem>Option 1</MenuItem>
+              <MenuItem>Option 2</MenuItem>
+              <MenuItem>Option 3</MenuItem>
+            </MenuList>
+          </MenuPopover>
+        </Menu>
+      </MenuList>
+    </MenuPopover>
+  <Menu>
+)
+```
+
+### Selection
+
+```tsx
+const [selectedItems, setSelectedItems] = React.useState([]);
+const menuCheckbox = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+  >
+    <MenuTrigger><Button>Open menu</Button></MenuTrigger>
+    <MenuPopover>
+      <MenuList>
+        <MenuItemCheckbox name="checkbox1">Option 1</MenuItemCheckbox>
+        <MenuItemCheckbox name="checkbox2">Option 2</MenuItemCheckbox>
+        <MenuItemCheckbox name="checkbox3">Option 3</MenuItemCheckbox>
+      </MenuList>
+    </MenuPopover>
+<Menu>
+)
+```
+
+## Visual Examples
+
+Coming soon
+
+## Variants
+
+### Nested menus
+
+A `Menu` should be able to trigger a submenu, or an additional instance of `Menu`, as a part of one or more of its options. The nested `Menu` component should have the same functional capabilities as the root `Menu` component, however it will have some default behavioral differences.
+
+Nested menus are by default triggered by hovering over a trigger, but can also be opened by invoking the trigger component for the nested `Menu`.
+
+### Selection state
+
+A `Menu` should be able to track and represent the selection, or checked, state of its options.
+
+When an option has a configurable selection state, it does not dismiss the `Menu` when toggled, as opposed to other options which close the `Menu` on being invoked by default.
+
+### Sections
+
+A `Menu` can be divided into sections using dividers. The `Menu` does not support `MenuGroups` yet, but can visually represent sections using `MenuDividers`.
+
+### Disabled option(s)
+
+All options in a `Menu` component can be disabled and should provide a visible indication for this purpose. Although disabled actions cannot be invoked, they can are keyboard focusable for the purposes of accessibility.
```

Comment:
"they can are keyboard focusable" - is `can are` part expected?

Reply:
Ah thanks for the catch